File: C++/1037 약수.cpp
#include <iostream>
using namespace std;

int main()
{
	int mini = 1000001;
	int maxi = 0;
	int n;
	int t;
	cin >> t;
	for (int i = 0; i < t; i++) {
		cin >> n;
		if (n < mini)
			mini = n;
		if (n > maxi)
			maxi = n;
	}

	cout << mini * maxi;
	
	return 0;
}

File: python/1037 약수.py
mini = 1000001
maxi = 0
t = int(input())
nums = list(map(int, input().split()))
for i in range(t):
    if nums[i] < mini:
        mini = nums[i]
    if nums[i] > maxi:
        maxi = nums[i]

print(mini * maxi)
